crates/gpui/src/text_system/line.rs

```diff
@@ -147,7 +156,17 @@ fn paint_line(
         let mut current_strikethrough: Option<(Point<Pixels>, StrikethroughStyle)> = None;
         let mut current_background: Option<(Point<Pixels>, Hsla)> = None;
         let text_system = cx.text_system().clone();
-        let mut glyph_origin = origin;
+        let mut glyph_origin = point(
+            aligned_origin_x(
+                origin,
+                align_width.unwrap_or(layout.width),
```

Comment:
Here `layout.width` is the actually text width, not the line width.

So if the text is "hello", the `layout.width` be the width of the "hello".